scipy/signal/_fir_filter_design.py

```diff
@@ -834,6 +835,296 @@ def remez(numtaps, bands, desired, *, weight=None, type='bandpass',
                             maxiter, grid_density)
 
 
+def _oddround(x):
+    """Return the nearest odd integer from x."""
+
+    return x - np.mod(x, 2) + 1
+
+
+def _oddceil(x):
+    """Return the smallest odd integer not less than x."""
+
+    return _oddround(x + 1)
+
+
+def _remlplen_herrmann(fp, fs, dp, ds):
+    """
+    Determine the length of the low pass filter with passband frequency
+    fp, stopband frequency fs, passband ripple dp, and stopband ripple ds.
+    fp and fs must be normalized with respect to the sampling frequency.
+    Note that the filter order is one less than the filter length.
+
+    Uses approximation algorithm described by Herrmann et al.:
+    """
+
+    dF = fs - fp
+    a = [5.309e-3, 7.114e-2, -4.761e-1, -2.66e-3, -5.941e-1, -4.278e-1]
+    b = [11.01217, 0.51244]
+    Dinf = (
+        np.log10(ds) * (a[0] * np.log10(dp) ** 2 + a[1] * np.log10(dp) + a[2])
+        + a[3] * np.log10(dp) ** 2
+        + a[4] * np.log10(dp)
+        + a[5]
+    )
+    f = b[0] + b[1] * (np.log10(dp) - np.log10(ds))
+    N1 = Dinf / dF - f * dF + 1
+
+    return int(_oddround(N1))
+
+
+def _remlplen_kaiser(fp, fs, dp, ds):
+    """
+    Determine the length of the low pass filter with passband frequency
+    fp, stopband frequency fs, passband ripple dp, and stopband ripple ds.
+    fp and fs must be normalized with respect to the sampling frequency.
+    Note that the filter order is one less than the filter length.
+
+    Uses approximation algorithm described by Kaiser:
+    """
+
+    dF = fs - fp
+    N2 = (-20 * np.log10(np.sqrt(dp * ds)) - 13.0) / (14.6 * dF) + 1.0
+
+    return int(_oddceil(N2))
+
+
+def _remlplen_ichige(fp, fs, dp, ds):
+    """
+    Determine the length of the low pass filter with passband frequency
+    fp, stopband frequency fs, passband ripple dp, and stopband ripple ds.
+    fp and fs must be normalized with respect to the sampling frequency.
+    Note that the filter order is one less than the filter length.
+
+    Uses approximation algorithm described by Ichige et al.
+    """
+    dF = fs - fp
+
+    def v(dF, dp):
+        return 2.325 * ((-np.log10(dp)) ** -0.445) * dF ** (-1.39)
+
+    def arctan_g(fp, dF, d):
+        return (2.0 / np.pi) * np.arctan(
+            v(dF, dp) * (1.0 / fp - 1.0 / (0.5 - dF))
+        )
+
+    def arctan_h(fp, dF, c):
+        return (2.0 / np.pi) * np.arctan(
+            (c / dF) * (1.0 / fp - 1.0 / (0.5 - dF))
+        )
+
+    Nc = ceil(1.0 + (1.101 / dF) * (-np.log10(2.0 * dp)) ** 1.1)
+    Nm = (0.52 / dF) * np.log10(dp / ds) * (-np.log10(dp)) ** 0.17
+    N3 = ceil(
+        Nc
+        * (arctan_g(fp, dF, dp) + arctan_g(0.5 - dF - fp, dF, dp) + 1.0)
+        / 3.0
+    )
+    DN = ceil(
+        Nm
+        * (
+            arctan_h(fp, dF, 1.1)
+            - (arctan_h(0.5 - dF - fp, dF, 0.29) - 1.0) / 2.0
+        )
+    )
+    N4 = N3 + DN
+
+    return int(N4)
+
+
+def remezord(freqs, amps, rips, fs=1.0, alg="ichige"):
```

Comment:
We should consider making some of these keyword only as per https://docs.scipy.org/doc/scipy/dev/missing-bits.html#required-keyword-names

Please feel free to suggest a better location of the `*`
```suggestion
def remezord(freqs, amps, rips, *, fs=1.0, alg="ichige"):
```